sdk/state/update.go

```diff
@@ -0,0 +1,124 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/stellar/experimental-payment-channels/sdk/txbuild"
+	"github.com/stellar/go/xdr"
+)
+
+type Payment struct {
+	IterationNumber       int64
+	Amount                int64
+	CloseSignatures       []xdr.DecoratedSignature
+	DeclarationSignatures []xdr.DecoratedSignature
+	FromInitiator         bool
+}
+
+func (c *Channel) NewPayment(amount int64) (*Payment, error) {
+	if amount <= 0 {
+		return nil, errors.New("payment amount must be greater than 0")
+	}
+	newBalance := int64(0)
+	if c.initiator {
+		newBalance = c.balance.Amount + amount
+	} else {
+		newBalance = c.balance.Amount - amount
+	}
+	txC, err := txbuild.Close(txbuild.CloseParams{
+		ObservationPeriodTime:      c.observationPeriodTime,
+		ObservationPeriodLedgerGap: c.observationPeriodLedgerGap,
+		InitiatorSigner:            c.initiatorSigner(),
+		ResponderSigner:            c.responderSigner(),
+		InitiatorEscrow:            c.initiatorEscrowAccount().Address,
+		ResponderEscrow:            c.responderEscrowAccount().Address,
+		StartSequence:              c.startingSequence,
+		IterationNumber:            c.iterationNumber,
+		AmountToInitiator:          maxInt64(0, newBalance*-1),
+		AmountToResponder:          maxInt64(0, newBalance),
+	})
+	if err != nil {
+		return nil, err
+	}
+	txC, err = txC.Sign(c.networkPassphrase, c.localSigner)
+	if err != nil {
+		return nil, err
+	}
+	return &Payment{
+		Amount:          amount,
+		CloseSignatures: txC.Signatures(),
+		FromInitiator:   c.initiator,
+	}, nil
+}
+
+func (c *Channel) ConfirmPayment(p *Payment) (*Payment, error) {
```

Comment:
We'll need to normalize the names of the functions. In my PR I used OpenPropose and OpenConfirm, so they're grouped by Open, but they read a little odd. NewOpen would also be odd though, there can only ever be one open and close.

What do you think of the following? The names Begin and Commit are from database terms, although it works a bit differently since you have to commit multiple times.

```
OpenBegin
OpenCommit

PaymentBegin
PaymentCommit

CloseBegin
CloseCommit
```